plugin/storage/storage.go

```diff
@@ -0,0 +1,11 @@
+package storage
+
+import (
+	"context"
+	"io"
+)
+
+// Uploader uploads contents to the cloud.
+type Uploader interface {
```

Comment:
Is this interface necessary?

Reply:
This is for the future GCS client. If we prefer to deal with that later, I can remove this interface and just pass in an s3 client struct .

Reply:
Let's deal with it later. We never know the best design of interfaces till we hit >3 types.

Reply:
SGTM. Changed.